### DISCUSSION TITLE
Component Proposal: Glossary

### DISCUSSION THREAD

**Discussion by debjudy** [uswds/uswds#5771] [Proposals]
Examples
https://design.va.gov/content-style-guide/word-list
https://www.fec.gov/help-candidates-and-committees/registering-political-party/

https://github.com/18F/glossary

https://www.usaspending.gov/

DESIGN ASSETS
- [ ] 10% - Collect and analyze examples;  draft research insights and list of features for team/client feedback
- [ ] 20% - Draft user story & clarify the problem(s) / user needs
- [ ] 30% - Low-fi design review
- [ ] 40% - Get community feedback on initial, Hi-fi designs
- [ ] 50% - Hi-fi Sketch assets review
- [ ] 60% - Hi-fi Adobe XD assets review
- [ ] 80% - Sketch and XD assets review w/ Dan + merged into master working files
- [ ] 100% - PR approved by Dan

GUIDANCE
- [ ] 10% - Summarize insights from research (desk + interviews)
- [ ] 20% - Draft guidance review no. 1 (minimum 2 bullets each section)
- [ ] 40% - Get community feedback on draft guidance
- [ ] 50% - Draft guidance review no. 2 (accessibility focus)
- [ ] 60% - Draft guidance review no. 3 (w/ Dan)
- [ ] 80% - Add component to research report and slide deck
- [ ] 100% - PR approved by Dan

CODE
- [ ] 10% - Code review no. 1 (Federalist preview) - interaction design focus
- [ ] 20% - Code review no. 2 (Federalist preview) - visual design focus
- [ ] 40% - Get community feedback or conduct tests with users, round 1
- [ ] 60% - Code review no. 3 (Federalist preview) - accessibility focus
- [ ] 80% - Get community feedback or conduct tests with users, round 2
- [ ] 100% - PR approved by Dan

**Comment by smcharg-ohss** [comment]
Here's another example: https://revenuedata.doi.gov/

When I was at Interior, we had the book icon like the FEC site linked above has and found during user interviews that people didn't always understand what the icon meant. Many didn't associate the book with getting more information or help and some thought it was a bible, so we switched to the question mark icon. Many more participants associated the question mark with getting more information about a term.

We would definitely use a glossary component at the Office of Homeland Security Statistics.

**Reply by mejiaj** [reply]
I see the tooltips explaining in the link you sent

<img src="https://github.com/uswds/uswds/assets/3385219/ad2cae42-ea5d-448a-bf64-5e4a9363c475" width="300" />

And I also see the top nav link to a **[Glossary](https://revenuedata.doi.gov/glossary)** page.

--- 

I can see the usefulness of combining the approaches like:

- A Glossary Box at the top of the page with terms used specifically on that page. 
- A link to view the full glossary terms via a drawer menu or something similar.

**Reply by smcharg-ohss** [reply]
That site used to have a drawer glossary, but it had accessibility issues with the entire drawer being read by screen readers before the contents on the page, so we moved the full glossary to a page.

**Comment by afeijoo** [comment]
Digital.gov has several glossaries, and we'd make heavy use of a glossary component. Examples:

* https://pra.digital.gov/
* https://digital.gov/guides/hcd/

**Comment by amycole501** [comment]
There are accessibility issues with some of these examples (especially ones with a hover-only option). 

That said,  there would be a huge benefit to having a component/method for explaining complex terms. We are researching this on a project where some American terms don't translate well in other languages. 

A complicating factor is taking someone off a page to a glossary section or distracting them with a widget that opens on the side with additional terms. Testing with people with ADHD for example, indicates they want to stay as focused on a single task as possible.

I was considering a method of keeping a person on a page and allowing them to "learn more" or explore a topic more deeply without a hover or tooltip interaction. The USWDS accordion could be repurposed into a "details" component.  See the [UK design system example.](https://design-system.service.gov.uk/components/details/)

**Reply by smcharg-ohss** [reply]
The Interior example works on both hover or click.

**Reply by amycole501** [reply]
Yes that tooltip is mostly keyboard accessible and is styled very nicely for readability.

However it's not [able to be dismissed ](https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html ) (when I tested keyboard tabbing into the DOI tooltip example I couldn't escape out of it) which is an issue we recently improved on the USWDS tooltip component. That update is coming soon for teams to implement.

Some tooltip style popups have shown to be problematic in our auditing since the font can sometimes be very small especially on mobile where they can also disappear too quickly before lengthier definitions are able to be read. 

Tooltips that are longer or include lengthy definitions can also obscure content around the element receiving focus. 

I could see how a tooltip could work well, too, but in any of these options we'd want to perform usability testing with a wide variety of users and devices.

**Comment by jaclinec** [comment]
Just wanted to post an update - we're currently evaluating this proposal against a few guiding acceptance criteria, such as:

- Does the component support a WCAG requirement? At what level (A, AA, AAA)?
- Does the component have a clear purpose and address a user need? 
- Is there an existing solution in USWDS that better meets the need than the proposed solution?
- Do the usability and accessibility benefits outweigh any potential risks of using the solution?
- To what extent is the solution needed by the USWDS community? 

We're wrapping up some desk research so we can make an evidence-based decision. So far, it's looking like it's ideal to use a combination of strategies for helping users understand unfamiliar terms. Those could include using a standalone glossary, but could also include point-of-need solutions like accordions, modals, and using plain language alternatives next to the technical terms in context. More to come!

**Comment by amycole501** [comment]
[Proposal process documented here](https://github.com/uswds/uswds-proposals/issues/40)

**Comment by ryparker** [comment]
I'm bringing [#6259](https://github.com/uswds/uswds/issues/6259) here so we have one place to discuss inline definitions. It proposes dash-underlined terms that show definitions in a tooltip; the original issue remains available for its history.

Before choosing that design, we need examples of when an inline definition helps more than a glossary page, and how people would find and use it with keyboard, touch, zoom, and screen readers. Please share examples or research here. Closing the separate issue doesn't resolve that need, and we haven't chosen a tooltip design or delivery date.
